src/topology/category/Top/limits.lean

```diff
@@ -125,6 +126,452 @@ instance forget_preserves_colimits : preserves_colimits (forget : Top.{u} ⥤ Ty
+-/
+def prod_iso_prod (X Y : Top.{u}) : X ⨯ Y ≅ Top.of (X × Y) :=
+(limit.is_limit _).cone_point_unique_up_to_iso (prod_binary_fan_is_limit X Y)
+
+@[simp, reassoc] lemma prod_iso_prod_hom_fst (X Y : Top.{u}) :
+  (prod_iso_prod X Y).hom ≫ ⟨prod.fst⟩ = limits.prod.fst :=
+by simpa [← iso.eq_inv_comp, prod_iso_prod]
+
+@[simp, reassoc] lemma prod_iso_prod_hom_snd (X Y : Top.{u}) :
+  (prod_iso_prod X Y).hom ≫ ⟨prod.snd⟩ = limits.prod.snd :=
+by simpa [← iso.eq_inv_comp, prod_iso_prod]
+
+@[simp] lemma prod_iso_prod_hom_apply {X Y : Top.{u}} (x : X ⨯ Y) :
+  (prod_iso_prod X Y).hom x =
+    ((limits.prod.fst : X ⨯ Y ⟶ _) x, (limits.prod.snd : X ⨯ Y ⟶ _) x) :=
+begin
+  ext,
+  { exact concrete_category.congr_hom (prod_iso_prod_hom_fst X Y) x },
+  { exact concrete_category.congr_hom (prod_iso_prod_hom_snd X Y) x }
+end
+
+@[simp, reassoc, elementwise] lemma prod_iso_prod_inv_fst (X Y : Top.{u}) :
+  (prod_iso_prod X Y).inv ≫ limits.prod.fst = ⟨prod.fst⟩ :=
+by simp [iso.inv_comp_eq]
+
+@[simp, reassoc, elementwise] lemma prod_iso_prod_inv_snd (X Y : Top.{u}) :
+  (prod_iso_prod X Y).inv ≫ limits.prod.snd = ⟨prod.snd⟩ :=
+by simp [iso.inv_comp_eq]
+
+lemma range_prod_map {W X Y Z : Top.{u}} (f : W ⟶ Y) (g : X ⟶ Z) :
+  set.range (limits.prod.map f g) =
+    (limits.prod.fst : Y ⨯ Z ⟶ _) ⁻¹' (set.range f) ∩
+      (limits.prod.snd : Y ⨯ Z ⟶ _) ⁻¹' (set.range g) :=
+begin
+  ext,
+  split,
+  { rintros ⟨y, rfl⟩,
+    simp only [set.mem_preimage, set.mem_range, set.mem_inter_eq, ←comp_apply],
+    simp only [limits.prod.map_fst, limits.prod.map_snd,
+      exists_apply_eq_apply, comp_apply, and_self] },
+  { rintros ⟨⟨x₁, hx₁⟩, ⟨x₂, hx₂⟩⟩,
+    use (prod_iso_prod W X).inv (x₁, x₂),
+    apply concrete.limit_ext,
+    rintro ⟨⟩,
+    { simp only [← comp_apply, category.assoc], erw limits.prod.map_fst, simp [hx₁] },
+    { simp only [← comp_apply, category.assoc], erw limits.prod.map_snd, simp [hx₂] } }
+end
+
+lemma inducing_prod_map {W X Y Z : Top} {f : W ⟶ X} {g : Y ⟶ Z}
+  (hf : inducing f) (hg : inducing g) : inducing (limits.prod.map f g) :=
+begin
+  constructor,
+  simp only [prod_topology, induced_compose, ←coe_comp, limits.prod.map_fst, limits.prod.map_snd,
+    induced_inf],
+  simp only [coe_comp],
+  rw [← @induced_compose _ _ _ _ _ f, ← @induced_compose _ _ _ _ _ g, ←hf.induced, ←hg.induced]
+end
+
+lemma embedding_prod_map {W X Y Z : Top} {f : W ⟶ X} {g : Y ⟶ Z}
+  (hf : embedding f) (hg : embedding g) : embedding (limits.prod.map f g) :=
+⟨inducing_prod_map hf.to_inducing hg.to_inducing,
+begin
+  haveI := (Top.mono_iff_injective _).mpr hf.inj,
+  haveI := (Top.mono_iff_injective _).mpr hg.inj,
+  exact (Top.mono_iff_injective _).mp infer_instance
+end⟩
+
+end prod
+
+section pullback
+
+lemma pullback_topology {X Y Z : Top} (f : X ⟶ Z) (g : Y ⟶ Z) :
+  (pullback f g).topological_space =
+    induced (pullback.fst : pullback f g ⟶ _) X.topological_space ⊓
+      induced (pullback.snd : pullback f g ⟶ _) Y.topological_space :=
```

Comment:
Again, I think the way to prove this is to follow what you did above for products and limits -- I.e. first make an explicit limit cone, then use the homeomorphism obtained from that to get this result.